Frends.ServiceBus/ServiceBus.cs

```diff
@@ -341,6 +348,109 @@ private static Encoding GetEncodingFromContentType(string contentTypeString, Enc
                 }
             }
         }
+
+        static async public Task<ReadResult> SendAndWaitForResponse(SendAndWaitForResponseInput input, SendAndWaitForResponseOptions options, CancellationToken token = new CancellationToken())
+        {
+            // Prepare queues
+            if (options.CreateQueues)
+            {
+                await EnsureQueueExists(input.Queue, input.ConnectionString, false);
+                await EnsureQueueExists(input.ReplyToQueue, input.ConnectionString, true);
+            }
+
+            QueueClient requestClient = MessagingFactory.CreateFromConnectionString(input.ConnectionString).CreateQueueClient(input.Queue);
+            QueueClient responseClient = MessagingFactory.CreateFromConnectionString(input.ConnectionString).CreateQueueClient(input.ReplyToQueue);
+            MessageSession session = responseClient.AcceptMessageSession(input.SessionID);
```

Comment:
Could the existing read task just do this, accept messages from a given session? E.g. by just giving the session ID  as a parameter, the same way as here. Then there would be no need for a separate task